### ISSUE TITLE
Feature Request - Modify Server Endpoint

### ISSUE THREAD

**Issue by thexmanxyz** [nextcloud/desktop#421]
Hey there,

I will start with this issue. We need an option to change the current server endpoint URL. Currently this is not possible and this can at least under Windows cause serious problems like I faced today but more on that in the next issue (#PLACEHOLDER).

I would really appreciate that feature because currently over GUI you can only remove the old entry and readd the new URL and reattach all folders. I later found out that you can change it via config however this might not be a huge effort but makes managing a lot easier.

**Comment by paskalito** [comment]
yes im also needing this. - due to a neccecary change of my servers url since the dyndns provider died.

i also did it without problems via the config, but for non tecsavy people thats just too much / nothing they would do..

**Comment by florianrusch** [comment]
@thexmanxyz and @paskalito: can you explain, how to change the domain/url of an attached server? I've searched for maybe an half hour but couldn't find anything :-/

**Comment by thexmanxyz** [comment]
@florianrusch As I said you can only do this in the [config file ](https://docs.nextcloud.com/desktop/2.3/advancedusage.html#configuration-file) - please check the link for more information in the docs.

> On Linux distributions:
$HOME/.config/Nextcloud/nextcloud.cfg

> On Microsoft Windows systems:
%APPDATA%\Nextcloud\nextcloud.cfg

> On macOS systems:
$HOME/Library/Preferences/Nextcloud/nextcloud.cfg

**Comment by florianrusch** [comment]
Ahhhh shit 😆 I thought I already looked at this file. 🙈 Thank you very much.

For everyone who can't find the file on MacOS:
have a look at `~Library/Application\ Support/Nextcloud/nextcloud.cfg`

**Comment by thexmanxyz** [comment]
@florianrusch you are welcome, however as I see the path is different than the one listed in the Docs but TBH I'm not a Mac guy. However I found out that the path changed during Owncloud / Nextcloud development (see docs) from:

-`~Library/Application\ Support/Nextcloud/nextcloud.cfg`
to
- `~Library/Preferences/Nextcloud/nextcloud.cfg`

So I'm not sure if this has something to do with the MacOS version or if you should better change the file listed in the newer docs (last one in the enumeration).

**Comment by florianrusch** [comment]
@thexmanxyz If I'm right, than is my nextcloud client installation older than this commit. Of course I'm on the latest version of the client.

I've tried to move the hole folder into `~Library/Preferences/Nextcloud/`. If I'm starting the client after the movement of the files, I'm getting the dialog where I should config my first server. It looks like the client doesn't check the `~Library/Preferences/Nextcloud/` folder. 🤔 

Looks for me like a bug?

**Comment by thexmanxyz** [comment]
@florianrusch If you are on the latest version of the client, the changes of the commit should actually be part of your version. Because the commit is 1 year old and the changed path is already quite long in the docs as well. It dates back a few release version. But TBH I don't know what's really going on here. I just noticed it, after you reported back that you modified the file in the old folder. Also there is no hint in the commit and the attached issue number why or for what reason the path has changed, so I'm completely clueless. But as it does work with `~Library/Application\ Support/Nextcloud/`, nevermind. 

Thanks that you tried out moving the folder and added your finding here because this probably helps other people. There might be more users searching for the problem or noticing the same issue.

Something that right now comes into my mind is, that the path might only change during a fresh installation, that could also be the reason in your case (through update the path is eventually kept like on initial installation - but this is a guess in the dark). But don't care too much. If this is the case, it should however be documented somewhere...

**Comment by PascalHeynmoeller** [comment]
### Is your feature request related to a problem? Please describe.
After an update of my Router which switchted from an letsencrypt protected topleveldomain to an full qualified domain name (hostname added), I can't change the fqdn in the account settings of the app to match the new name.
### Describe the solution you'd like
Make the hostname of the server for an specific account in the Nextcloud app editable (like in SFOS or by settings editor in thunderbird).
### Describe alternatives you've considered
The alternative is to use nextcloudcmd instead of gui or create an new conf file (as the conf file doesn't exist by default). Or create new profile and sync everything again, what we try to avoid.
### Additional context
AVM Fritz!Box routers now supports multiple hosts with Letsencrypt so using an fqdn with hostname instead of topleveldomain only is now how the Router works.

**Comment by Rello** [comment]
The client has improved a lot since the ticket was opened.
If it is still relevant, please create a fresh ticket.
Thank you for your support
